### PR TITLE
PHPORM-153: Add automated release workflow

### PR DESCRIPTION
**Pull request by alcaeus**
PHPORM-153

The release workflow is an exact copy of the one used in PHPLIB, with the necessary adjustments to the release message.

The necessary secrets have been added to the repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  GIT_AUTHOR_NAME: "DBX PHP Release Bot"
+  GIT_AUTHOR_EMAIL: "dbx-php@mongodb.com"
+  default-release-message: |
+    The PHP team is happy to announce that version {0} of the MongoDB integration for Laravel is now available.
+
+    **Release Highlights**
+
+    TODO: one or more paragraphs describing important changes in this release
+
+    A complete list of resolved issues in this release may be found in [JIRA](https://jira.mongodb.org/secure/ReleaseNote.jspa?version={1}&projectId=22488).
```
Comment (jmikola) [inline .github/workflows/release.yml]: This seems like something to change given the [existing releases](https://github.com/mongodb/laravel-mongodb/releases). I'm not familiar with how "Release Highlights" are generated, but we definitely don't link to JIRA. We _do_ have a link for "Full Changelog", though.

**Review by jmikola — APPROVED** [review]
Actual changes LGTM but I'll defer to you and @GromNaN regarding the release template.

File: .github/workflows/release.yml (hunk 1 above)
Comment (GromNaN) [reply .github/workflows/release.yml]: I'd prefer to use GitHub milestones for that.
The full git history is better for now.

Comment (alcaeus) [reply .github/workflows/release.yml]: Re-added automatic release note generation and removed our own release notes template. We'll [customise release note generation](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuring-automatically-generated-release-notes) to exclude merge-up pull requests and categorise other pull requests.

**Review by GromNaN — APPROVED** [review]

**Pull request merged**